client/src/app/pages/archetypes/components/archetype-detail-drawer.tsx

```diff
@@ -164,6 +168,21 @@ const ArchetypeDetailDrawer: React.FC<IArchetypeDetailDrawerProps> = ({
   );
 };
 
+const ApplicationLabels: React.FC<{ applicationRefs?: Ref[] }> = ({
+  applicationRefs,
+}) =>
+  (applicationRefs?.length ?? 0) === 0 ? null : (
```

Comment:
With the length wrapper already in the description list, `applicationRefs` should be required and this step can be skipped.